gui/common_qt_functions.py

```diff
@@ -102,6 +121,16 @@ def setComponentSettings(self):
             'buffer-size-mb': 0,
             'file-caching': False # false = handle level caching ON
         })
+        self.settings.setValue('block_cache',{
```

Comment:
Adding block_cache might be enough of a major change to justify adding components to the GUI page to reflect this.

Reply:
We can definitely add GUI components for the block cache, but I think that can be a separate PR.
There may be an issue with the GUI removing block_cache from a pipeline written by the user.
I've added a story for these questions.
For now, I'd like to disjoint the GUI's YAML validation from its actual settings interface. I've added a story for re-evaluating which options should be exposed.

Reply:
I added `block_cache` to the dropdown menu in S3 and Azure settings, and added it to the list of possible components.
I did a quick run test, and it works (the drop-down shows up, it doesn't clobber the config, and it loads from the config correctly).

Reply:
Nice job